app/jobs/import_from_vacancy_source_job.rb

```diff
@@ -0,0 +1,84 @@
+class ImportFromVacancySourceJob < ApplicationJob
+  queue_as :low
+
+  def perform(source_klass)
+    return if DisableExpensiveJobs.enabled?
+
+    @source_name = source_klass.source_name
+    import_from_source(source_klass)
+  end
+
+  private
+
+  def import_from_source(source_klass)
+    total_count = 0
+    source_external_references = []
+    errors = {}
+
+    source_klass.new.each do |vacancy|
+      total_count += 1
+      source_external_references << vacancy.external_reference
+
+      # Avoids unnecessary and expensive DB updates for vacancies that have not changed since they were imported.
+      next if vacancy.persisted? && !vacancy.changed?
```

Comment:
This is the main performance update. It doesn't even try to import it if the assigned attributes into the existing vacancy didn't change.